packages/playground-examples/copy/pt/JavaScript/Functions with JavaScript/Generic Functions.ts

```diff
@@ -0,0 +1,100 @@
+// Tipos genéricos provêm uma forma de usar Tipos como variáveis em outros
+// tipos. Meta.
+
+// Tentaremos manter este exemplo leve. Você pode fazer
+// muita coisa com tipos genéricos e é provável que veja algum código bem
+// complicado usando tipos genéricos em algum ponto - mas isso
+// não significa que tipos genéricos são complicados.
+
+// Vamos começar com um exemplo onde envolveremos um objeto de entrada
+// em um array. Apenas nos importaremos com uma só variável neste
+// caso: o tipo que foi passado como argumento:
+
+function envolverEmArray<Tipo>(entrada: Tipo): Tipo[] {
+  return [entrada];
+}
+
+// Nota: é comum ver Tipo ser referido como T. Isso é
+// culturalmente similar à como as pessoas usam i em um loop for
+// para representar índice. T normalmente representa Tipo, então
+// usaremos o nome completo para maior clareza
+
+// Nossa função usará inferência para sempre manter o tipo
+// passado como argumento igual ao tipo retornado (porém
+// envolvido em um array)
+
+const stringArray = envolverEmArray("hello generics");
+const numberArray = envolverEmArray(123);
+
+// Podemos verificar que isso funciona como esperado ao checar
+// se podemos atribuir um array de strings à uma função que
```

Comment:
Grammar is incorrect (a crase não se aplica). The proper format is "... atribuir um array de strings a uma ..."